text/0000-release-channels.md

```diff
@@ -0,0 +1,471 @@
+  `stable`, but features may be partially deprecated. This ensures
+  that no APIs are accidentally excluded from stabilization and that
+  entire features may be considered either 'unstable' or 'stable'.
+
+It's important to note that these stability attributes are *only known
+to be useful to the standard distribution*, because of the explicit
+linkage to language versions and release channels. There is though no
+mechanism to explicitly forbid their use outside of the standard
+distribution. A general mechanism for indicating API stability
+will be reconsidered in the future.
+
+### API lifecycle
+
+These attributes alter the process of how new APIs are added to the
+standard library slightly. First an API will be proposed via the RFC
+process, and a name for the API feature being added will be assigned
+at that time. When the RFC is accepted, the API will be added to the
+standard library with an `#[staged_unstable(feature =
+"...")]`attribute indicating what feature the API was assigned to.
+
+After receiving test coverage from nightly users (who have opted into
+the feature) or thorough review, all APIs with a given feature will be
+changed from `staged_unstable` to `staged_stable`, adding `since =
+"..."` to mark the version in which the promotion occurred, and the
+feature is considered stable and may be used on the stable release
+channel.
+
+When a stable API becomes deprecated the `staged_deprecated` attribute
+is added in addition to the existing `staged_stable` attribute, as
+well recording the version in which the deprecation was performed with
+the `since` parameter.
+
+(Occassionally unstable APIs may be deprecated for the sake of easing
+user transitions, in which case they receive both the `staged_stable`
+and `staged_deprecated` attributes at once.)
+
+### Checking `#[feature]`
+
+The names of features will no longer be a hardcoded list in the compiler
+due to the free-form nature of the `#[staged_unstable]` feature names.
+Instead, the compiler will perform the following steps when inspecting
+`#[feature]` attributes lists:
+
+1. The compiler will discover all `#![feature]` directives
+   enabled for the crate and calculate a list of all enabled features.
+2. While compiling, all unstable language features used will be
+   removed from this list. If a used feature is not enabled, then an
+   error is generated.
+3. A new pass, the stability pass, will be extracted from the current
+   stability lint pass to detect usage of all unstable APIs. If an
+   unstable API is used, an error is generated if the feature is not
+   used, and otherwise the feature is removed from the list.
+4. If the remaining list of enabled features is not empty, then the
+   features were not used when compiling the current crate. The compiler
+   will generate an error in this case unconditionally.
+
+These steps ensure that the `#[feature]` attribute is used exhaustively
+and will check unstable language and library features.
```

Comment:
I may be reading this wrong, but doesn't it mean that when a feature becomes stable, code that used it will have to be changed to remove the ``#[feature]`` attribute? What's the motivation for that!